modules/rostests/apitests/user32/GetClassInfo.c

```diff
@@ -0,0 +1,161 @@
+/*
+ * PROJECT:     ReactOS api tests
+ * LICENSE:     GPL-2.0+ (https://spdx.org/licenses/GPL-2.0+)
+ * PURPOSE:     Tests for GetClassInfo
+ * COPYRIGHT:   Copyright 2023 Timo Kreuzer <timo.kreuzer@reactos.org>
+ */
+
+#include "precomp.h"
+
+VOID Test_Desktop(VOID)
+{
+    WNDCLASSEXW wcex;
+    BOOL result;
+
+    memset(&wcex, 0xab, sizeof(wcex));
+
+    result = GetClassInfoExW(GetModuleHandle(NULL), (LPCWSTR)WC_DESKTOP, &wcex);
+    ok_int(result, (ULONG_PTR)WC_DESKTOP);
+
+    ok_hex(wcex.cbSize, 0xabababab);
+    ok_hex(wcex.style, 0x8);
+    ok(wcex.lpfnWndProc != NULL, "lpfnWndProc shound't be NULL\n");
+    ok_int(wcex.cbClsExtra, 0);
+    ok_int(wcex.cbWndExtra, 0);
+    ok_ptr(wcex.hInstance, GetModuleHandle(NULL));
+    ok_ptr(wcex.hIcon, NULL);
+    ok(wcex.hCursor != NULL, "hCursor shound't be NULL\n");
+    ok_ptr(wcex.hbrBackground, (HBRUSH)(ULONG_PTR)2);
+    ok_ptr(wcex.lpszMenuName, NULL);
+    ok_ptr(wcex.lpszClassName, (LPCWSTR)WC_DESKTOP);
+    ok_ptr(wcex.hIconSm, NULL);
+}
+
+VOID Test_Dialog(VOID)
+{
+    WNDCLASSEXW wcex;
+    BOOL result;
+
+    memset(&wcex, 0xab, sizeof(wcex));
+
+    result = GetClassInfoExW(GetModuleHandle(NULL), (LPCWSTR)(ULONG_PTR)WC_DIALOG, &wcex);
```

Comment:
the WC_DIALOG needs the extra ULONG_PTR but not WC_DESKTOP above?

Reply:
No, the desktop uses 2 LONGs